### PR TITLE
Feature/sumedh

### PR DESCRIPTION
**Pull request by sumedh-mohod**
file added

### CODE REVIEW

File: app/Http/Controllers/TestData.php [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    public function store(Request $request)
+    {
+        $request->validate([
+            // 'email' => 'required|email',
+            // 'phone' => 'required|size:10|digits:10',
+            // 'address' => 'required|min:10|max:100',
+            // ssc validation code
+            'ssc_school'=>'required|min:5|max:20',
+            'ssc_board'=>'required|min:5|max:20',
+            'ssc_percentage'=>'required|size:2|digits:2',
```
Comment (ithqindia) [inline app/Http/Controllers/TestData.php:17]: Percentage can be in 4/2 digits.
Remove this and instead use `'required|numeric|between:0,99.99',`

File: app/Http/Controllers/TestData1.php [hunk 2]
```diff
@@ -0,0 +1,34 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
+use App\Models\Hsc;
+class TestData1 extends Controller
```
Comment (ithqindia) [inline app/Http/Controllers/TestData1.php:7]: TestData1 -> EducationalInformationHsc

File: app/Http/Controllers/TestData2.php [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
+use App\Models\Graduation;
+class TestData2 extends Controller
+{
+    public function store(Request $request)
+    {
+        $request->validate([
+            // 'email' => 'required|email',
```
Comment (ithqindia) [inline app/Http/Controllers/TestData2.php:12]: Remove commented lines

File: app/Http/Controllers/TestData3.php [hunk 4]
```diff
@@ -0,0 +1,34 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
+use App\Models\Diploma;
+class TestData3 extends Controller
```
Comment (ithqindia) [inline app/Http/Controllers/TestData3.php:7]: TestData3 -> EducationalInformationDiploma

File: app/Http/Controllers/TestData4.php [hunk 5]
```diff
@@ -0,0 +1,38 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
+use App\Models\Post_Graduation;
+class TestData4 extends Controller
```
Comment (ithqindia) [inline app/Http/Controllers/TestData4.php:7]: EducationalInformationPostGraduation

File: app/Models/Post_Graduation.php [hunk 6]
```diff
@@ -0,0 +1,13 @@
+<?php
+
+namespace App\Models;
+
+use Illuminate\Database\Eloquent\Factories\HasFactory;
+use Illuminate\Database\Eloquent\Model;
+
+class Post_Graduation extends Model
```
Comment (ithqindia) [inline app/Models/Post_Graduation.php:8]: Post_Graduation -> PostGraduation

File: database/migrations/2021_12_13_051229_create_ssc.php [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+class CreateSsc extends Migration
+{
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('ssc', function (Blueprint $table) {
```
Comment (ithqindia) [inline database/migrations/2021_12_13_051229_create_ssc.php:16]: `sscs` We have to following naming convention

File: database/migrations/2021_12_13_060747_create_graduation.php [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+class CreateGraduation extends Migration
+{
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('graduation', function (Blueprint $table) {
```
Comment (ithqindia) [inline database/migrations/2021_12_13_060747_create_graduation.php:16]: graduations

File: database/migrations/2021_12_13_063357_create_diploma.php [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+class CreateDiploma extends Migration
+{
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('diploma', function (Blueprint $table) {
```
Comment (ithqindia) [inline database/migrations/2021_12_13_063357_create_diploma.php:16]: diplomas

File: database/migrations/2021_12_13_065726_create_post_graduation.php [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+class CreatePostGraduation extends Migration
+{
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('post_graduation', function (Blueprint $table) {
```
Comment (ithqindia) [inline database/migrations/2021_12_13_065726_create_post_graduation.php:16]: post_graduations

File: database/migrations/2021_12_13_071548_create_hsc.php [hunk 11]
```diff
@@ -0,0 +1,38 @@
...
+class CreateHsc extends Migration
+{
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('hsc', function (Blueprint $table) {
```
Comment (ithqindia) [inline database/migrations/2021_12_13_071548_create_hsc.php:16]: hscs

File: routes/web.php [hunk 12]
```diff
@@ -16,3 +24,29 @@
...
+
+
+Route::get('/education-details', function () {
+    return view('education-details');
+});
+
+
+
+Route::post('/education-details', [TestData::class, 'store']);
+Route::post('/testhsc', [TestData1::class, 'store']);
```
Comment (ithqindia) [inline routes/web.php:48]: /hsc-details

**Review by ithqindia — CHANGES_REQUESTED** [review]
Please format all your files and remove empty lines keep it to minimum one empty line.

**Comment by ithqindia** [comment]
Please format all your files and remove empty lines keep it to minimum one empty line.

File: app/Models/Diploma.php [hunk 13]
```diff
@@ -0,0 +1,11 @@
+<?php
+namespace App\Models;
+use Illuminate\Database\Eloquent\Factories\HasFactory;
+use Illuminate\Database\Eloquent\Model;
+
+class Diploma extends Model
+{
+    use HasFactory;
+    protected $table = "diploma";
```
Comment (ithqindia) [inline app/Models/Diploma.php:9]: protected $table = "diplomas";

File: app/Models/Graduation.php [hunk 14]
```diff
@@ -0,0 +1,11 @@
+<?php
+namespace App\Models;
+use Illuminate\Database\Eloquent\Factories\HasFactory;
+use Illuminate\Database\Eloquent\Model;
+
+class Graduation extends Model
+{
+    use HasFactory;
+    protected $table = "graduation";
```
Comment (ithqindia) [inline app/Models/Graduation.php:9]: protected $table = "graduations";

File: app/Models/Hsc.php [hunk 15]
```diff
@@ -0,0 +1,11 @@
+<?php
+namespace App\Models;
+use Illuminate\Database\Eloquent\Factories\HasFactory;
+use Illuminate\Database\Eloquent\Model;
+
+class Hsc extends Model
+{
+    use HasFactory;
+    protected $table = "hsc";
```
Comment (ithqindia) [inline app/Models/Hsc.php:9]: protected $table = "hscs";

File: app/Models/Post_Graduation.php [hunk 16]
```diff
@@ -0,0 +1,11 @@
+<?php
+namespace App\Models;
+use Illuminate\Database\Eloquent\Factories\HasFactory;
+use Illuminate\Database\Eloquent\Model;
+
+class Post_Graduation extends Model
+{
+    use HasFactory;
+    protected $table = "post_graduation";
```
Comment (ithqindia) [inline app/Models/Post_Graduation.php:9]: protected $table = "post_graduations";

File: app/Models/Ssc.php [hunk 17]
```diff
@@ -0,0 +1,11 @@
+<?php
+namespace App\Models;
+use Illuminate\Database\Eloquent\Factories\HasFactory;
+use Illuminate\Database\Eloquent\Model;
+
+class Ssc extends Model
+{
+    use HasFactory;
+    protected $table = "ssc";
```
Comment (ithqindia) [inline app/Models/Ssc.php:9]: protected $table = "sscs";

File: app/Models/Post_Graduation.php (hunk 6 above)
Comment (ithqindia) [reply app/Models/Post_Graduation.php:6]: Still not done

Comment (ithqindia) [reply app/Models/Post_Graduation.php:6]: change Post_Graduation -> PostGraduation and change file name PostGraduation.php

**Review by ithqindia — CHANGES_REQUESTED** [review]

File: app/Models/User.php [hunk 18]
```diff
@@ -41,4 +41,14 @@ class User extends Authenticatable
...
+
+    public function user_information()
+    {
+        // This needs to be Model name
+        return $this->hasOne(Ssc::class);
+        return $this->hasOne(Hsc::class);
+        return $this->hasOne(Graduation::class);
+        return $this->hasOne(Diploma::class);
+        return $this->hasOne(PostGraduation::class);
+    }
```
Comment (amolswnz) [inline app/Models/User.php:53]:
```
    public function ssc_information()
    {
        return $this->hasOne(Ssc::class);
    }

    public function Hsc_information()
    {
        return $this->hasOne(Hsc::class);
    }

    public function graduation_information()
    {
        return $this->hasOne(Graduation::class);
    }

    public function diploma_information()
    {
        return $this->hasOne(Diploma::class);
    }

    public function post_graduation_information()
    {
        return $this->hasOne(PostGraduation::class);
    }
```

Comment (ithqindia) [inline app/Models/User.php:53]: public function ssc_information()
    {
        return $this->hasOne(Ssc::class);
    }

    public function Hsc_information()
    {
        return $this->hasOne(Hsc::class);
    }

    public function graduation_information()
    {
        return $this->hasOne(Graduation::class);
    }

    public function diploma_information()
    {
        return $this->hasOne(Diploma::class);
    }

    public function post_graduation_information()
    {
        return $this->hasOne(PostGraduation::class);
    }

File: app/Http/Controllers/EducationalInformationDiplomaController.php [hunk 19]
```diff
@@ -0,0 +1,71 @@
...
+            $diplomaInformation->diploma_langauge = $request->get('diploma_langauge');
+            $diplomaInformation->diploma_year_form = $request->get('diploma_year_form');
+            $diplomaInformation->diploma_year_to = $request->get('diploma_year_to');
+            $diplomaInformation->save();
+        return view('success');
+    }
+
+    public function view(Request $request)
+    {
+        $user = User::with('user_information')->find($request->id);
```
Comment (ithqindia) [inline app/Http/Controllers/EducationalInformationDiplomaController.php:68]: $user = User::with('diploma_information')->find($request->id);

File: app/Http/Controllers/EducationalInformationGraduationController.php [hunk 20]
```diff
@@ -0,0 +1,71 @@
...
+            $graduationInformation->graduation_langauge = $request->get('graduation_langauge');
+            $graduationInformation->graduation_year_from = $request->get('graduation_year_from');
+            $graduationInformation->graduation_year_to = $request->get('graduation_year_to');
+            $graduationInformation->save();
+        return view('success');
+    }
+
+    public function view(Request $request)
+    {
+        $user = User::with('user_information')->find($request->id);
```
Comment (ithqindia) [inline app/Http/Controllers/EducationalInformationGraduationController.php:68]: change this as per above comment

File: app/Http/Controllers/EducationalInformationHscController.php [hunk 21]
```diff
@@ -0,0 +1,70 @@
...
+            $hscInformation->hsc_major = $request->get('hsc_major');
+            $hscInformation->hsc_langauge = $request->get('hsc_langauge');
+            $hscInformation->hsc_year_form = $request->get('hsc_year_form');
+            $hscInformation->hsc_year_to = $request->get('hsc_year_to');
+            $hscInformation->save();
+        return view('success');
+    }
+    public function view(Request $request)
+    {
+        $user = User::with('user_information')->find($request->id);
```
Comment (ithqindia) [inline app/Http/Controllers/EducationalInformationHscController.php:67]: change this as per above comment

File: app/Http/Controllers/EducationalInformationPostgraduationController.php [hunk 22]
```diff
@@ -0,0 +1,70 @@
...
+            $postGraduationInformation->post_graduation_langauge = $request->get('post_graduation_langauge');
+            $postGraduationInformation->post_graduation_year_from = $request->get('post_graduation_year_from');
+            $postGraduationInformation->post_graduation_year_to = $request->get('post_graduation_year_to');
+            $postGraduationInformation->save();
+        return view('success');
+    }
+
+    public function view(Request $request)
+    {
+        $user = User::with('user_information')->find($request->id);
```
Comment (ithqindia) [inline app/Http/Controllers/EducationalInformationPostgraduationController.php:67]: change this as per above comment

File: app/Http/Controllers/EducationalInformationSscController.php [hunk 23]
```diff
@@ -0,0 +1,73 @@
...
+            $sscInformation->ssc_langauge = $request->get('ssc_langauge');
+            $sscInformation->school_year_form = $request->get('school_year_form');
+            $sscInformation->school_year_to = $request->get('school_year_to');
+            $sscInformation->save();
+        return view('success');
+    }
+
+    public function view(Request $request)
+    {
+        $user = User::with('user_information')->find($request->id);
```
Comment (ithqindia) [inline app/Http/Controllers/EducationalInformationSscController.php:67]: change this as per above comment

**Review by ithqindia — APPROVED** [review]

File: resources/views/diploma-info.blade.php [hunk 24]
```diff
@@ -1,17 +1,107 @@
...
+    @if (isset($user->diploma_information))
+        <div class=" card mb-5 mb-xl-10" id="kt_profile_details_view">
+            <div class="card-header cursor-pointer">
+                <div class="card-title m-0">
+                    <h3 class="fw-bolder m-0">Diploma User Information</h3>
+                </div>
+            </div>
+            <div class="card-body p-9">
+                <div class="row mb-7">
+                    <lable class="col-lg-4 fw-bold text-muted">College Name</lable>
```
Comment (manii283) [inline resources/views/diploma-info.blade.php:58]: make the 2nd letter in lower case

**Pull request closed without merging**